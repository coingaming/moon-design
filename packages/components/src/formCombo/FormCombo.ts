import styled from 'styled-components';

const FormCombo = styled.div({
  display: 'block',
  position: 'relative',
});

export default FormCombo;
