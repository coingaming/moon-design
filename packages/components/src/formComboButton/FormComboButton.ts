import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';

import { inputSpacingY } from '../private/input/settings';

const FormComboButton = styled.div(({ theme }) => ({
  position: 'absolute',
  right: rem(inputSpacingY(theme) / 2),
  top: rem(inputSpacingY(theme) / 2),
}));

export default FormComboButton;
