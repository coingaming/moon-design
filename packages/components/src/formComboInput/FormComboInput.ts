import styled from 'styled-components';
import { rem } from '@heathmont/moon-utils';

import { Input } from '../private/input/input';

const FormComboInput = styled.div(({ theme: { radius } }) => ({
  [`${Input}`]: {
    borderRadius: rem(radius.largest),
  },
}));

export default FormComboInput;
