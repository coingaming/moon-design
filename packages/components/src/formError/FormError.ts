import styled from 'styled-components';
import { rem } from '@heathmont/sportsbet-utils';

const FormError = styled.p(({ theme: { color, space } }) => ({
  display: 'inline-block',
  marginTop: rem(space.small),
  fontSize: rem(12),
  color: color.chiChi[100],
}));

FormError.defaultProps = {
  role: 'alert',
};

export default FormError;
