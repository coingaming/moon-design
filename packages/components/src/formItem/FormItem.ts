import styled from 'styled-components';

const FormItem: any = styled.li({});

export default FormItem;
