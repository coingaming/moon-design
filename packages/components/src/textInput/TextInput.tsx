import * as React from 'react';
import styled from 'styled-components';

import { Label } from '../private/label/label';

import { Input } from './Input';
import { InputError } from './Error';

/**
 * Types & Settings
 */
type TextInputTypes =
  | 'date'
  | 'datetime-local'
  | 'email'
  | 'number'
  | 'password'
  | 'search'
  | 'tel'
  | 'text'
  | 'url'
  | 'time'
  | 'url';

type TextInputProps = React.InputHTMLAttributes<HTMLInputElement> & {
  label?: string;
  type?: TextInputTypes;
  placeholder?: string;
  error?: boolean | string;
  rounded?: boolean;
};

const TextInputElem = styled(Input as any)(({ error, theme: { color } }) => ({
  '&:focus': {
    borderColor: !error ? color.piccolo[100] : color.chiChi[100],
    outline: 'none',
  },
}));

/**
 * Component
 *
 * 1. Leaving the placeholder as an empty string by default allows us to float
 *    the label when a user starts typing, even if a placeholder isn't defined.
 */
const TextInput: React.FC<TextInputProps> = ({
  type = 'text',
  disabled,
  placeholder = ' ',
  label,
  error,
  rounded,
  ...props
}) => {
  const inputProps = {
    disabled,
    type,
    placeholder,
    rounded,
    ...props,
  };

  const TextInputInner = () => (
    <TextInputElem rounded={!!rounded} error={!!error} {...inputProps} />
  );

  if (!label) {
    return TextInputInner();
  }

  return (
    <Label text={label} disabled={disabled}>
      <>
        {TextInputInner()}
        {error && <InputError>{error}</InputError>}
      </>
    </Label>
  );
};

export { TextInputProps };

export default TextInput;
