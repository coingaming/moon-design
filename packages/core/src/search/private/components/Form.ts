import styled from 'styled-components';
import { rem } from '@heathmont/moon-utils';

import { SearchInputActive } from './Input';
import { Results, resultsActive } from './Results';
import { zIndex } from './settings';

/**
 * Search Form
 * The functional container for the search component.
 * We aren't using Design System form because we use focus-within logic here
 */
export const SearchForm = styled.form(({ theme }) => ({
  fontSize: rem(theme.space.default),
  position: 'relative',
  overflow: 'visible',
  width: '100%',
  [`&:focus-within, ${SearchInputActive}`]: {
    zIndex: zIndex.searchInputActive,
    [`~ ${Results}:not(:empty)`]: resultsActive(theme),
  },
}));
