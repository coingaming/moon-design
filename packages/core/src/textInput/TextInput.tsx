import React, { forwardRef } from 'react';
import { ColorProps } from '@heathmont/moon-themes';
import Size from '../private/enums/Size';
import TextInputBasic from './private/TextInputBasic';
import TextInputInnerLabel from './private/TextInputInnerLabel';
import TextInputPasswort from './private/TextInputPasswort';
import TextInputSizeType from './private/types/SizeTypes';
import TextInputTypes from './private/types/TextInputTypes';

export interface TextInputProps
  extends React.InputHTMLAttributes<HTMLInputElement> {
  id?: string;
  inputSize: TextInputSizeType | string;
  type: TextInputTypes | string;
  label?: JSX.Element | string;
  placeholder?: string;
  hintText?: JSX.Element | string;
  isError?: boolean;
  dir?: 'ltr' | 'rtl' | 'auto';
  showPasswordText?: JSX.Element | string;
  backgroundColor?: ColorProps;
  isSharpLeftSide?: boolean;
  isSharpRightSide?: boolean;
  isSharpTopSide?: boolean;
  isSharpBottomSide?: boolean;
  isTopBottomBorderHidden?: boolean;
  isSideBorderHidden?: boolean;
}

const TextInput = forwardRef<HTMLInputElement, TextInputProps>((props, ref) => {
  const { inputSize, type } = props;

  // render input with show/hide password
  if (type === 'password') {
    return <TextInputPasswort {...props} ref={ref} />;
  }
  if (inputSize === Size.XLARGE) {
    return <TextInputInnerLabel {...props} ref={ref} />;
  }
  // render rest sizes
  return <TextInputBasic {...props} ref={ref} />;
});

export default TextInput;
