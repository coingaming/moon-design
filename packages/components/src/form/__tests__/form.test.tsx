import * as React from 'react';
import { create } from 'react-test-renderer';
import { sportsbetDark, ThemeProvider } from '@heathmont/sportsbet-themes';
import 'jest-styled-components';

import Form from '../Form';
import FormItem from '../../formItem/FormItem';
import FormError from '../../formError/FormError';
import FormCombo from '../../formCombo/FormCombo';
import FormComboInput from '../../formComboInput/FormComboInput';
import FormComboButton from '../../formComboButton/FormComboButton';
import TextInput from '../../textInput/TextInput';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={sportsbetDark}>{component}</ThemeProvider>
);

const TestFormItems = () => (
  <React.Fragment>
    <FormItem>
      <label htmlFor="testInput1">
        Name
        <input id="testInput1" type="text" placeholder="e.g. John Appleseed" />
      </label>
    </FormItem>
    <FormItem>
      <label htmlFor="testInput1">
        Password
        <input id="testInput1" type="password" />
      </label>
      <FormError>Please enter a valid password.</FormError>
    </FormItem>
    <FormItem>
      <label htmlFor="testInput2">
        <input id="testInput2" type="checkbox" name="terms" value="terms" />I
        agree to the terms and conditions.
      </label>
    </FormItem>
    <FormItem>
      <button type="submit" value="submit">
        Sign up
      </button>
    </FormItem>
  </React.Fragment>
);

describe('Form', () => {
  test('renders correctly', () => {
    const form = create(
      renderWithTheme(
        <Form legend="Sign up">
          <TestFormItems />
        </Form>
      )
    );

    expect(form).toMatchSnapshot();
  });

  test('renders with HTML form attributes', () => {
    const form = create(
      renderWithTheme(
        <Form legend="Sign up" action="#" method="post" autocomplete="on">
          <TestFormItems />
        </Form>
      )
    );

    expect(form).toMatchSnapshot();
  });

  test('renders with custom maxWidth', () => {
    const form = create(
      renderWithTheme(
        <Form legend="Sign up" maxWidth="30rem">
          <TestFormItems />
        </Form>
      )
    );

    expect(form).toMatchSnapshot();
  });

  test('renders with custom fullWidth', () => {
    const form = create(
      renderWithTheme(
        <Form legend="Sign up" fullWidth>
          <TestFormItems />
        </Form>
      )
    );

    expect(form).toMatchSnapshot();
  });

  test('renders a form combo', () => {
    const form = create(
      renderWithTheme(
        <Form legend="2FA" maxWidth="25rem">
          <FormItem>
            <FormCombo>
              <FormComboInput>
                {/* We apply styles to specific childen */}
                <TextInput
                  label="Enable Two Factor Authentication"
                  placeholder="Enter code"
                />
              </FormComboInput>
              <FormComboButton>
                {/* We don't apply any styles to specific childen */}
                <button type="submit">Submit</button>
              </FormComboButton>
            </FormCombo>
          </FormItem>
        </Form>
      )
    );

    expect(form).toMatchSnapshot();
  });

  test('renders a form combo without a label', () => {
    const form = create(
      renderWithTheme(
        <Form legend="2FA" maxWidth="25rem">
          <FormItem>
            <FormCombo>
              <FormComboInput>
                {/* We apply styles to specific childen */}
                <TextInput placeholder="Enter code" />
              </FormComboInput>
              <FormComboButton>
                {/* We don't apply any styles to specific childen */}
                <button type="submit">Submit</button>
              </FormComboButton>
            </FormCombo>
          </FormItem>
        </Form>
      )
    );

    expect(form).toMatchSnapshot();
  });
});
