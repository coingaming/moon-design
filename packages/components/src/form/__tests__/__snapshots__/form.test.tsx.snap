// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders a form combo 1`] = `
.emotion-8 {
  border: 0;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-7 {
  display: block;
  margin-left: 0;
  margin-top: 0;
}

.emotion-6 {
  display: block;
  list-style-type: none;
  margin-bottom: 1rem;
}

.emotion-6:before {
  position: absolute;
  content: "\\200B";
}

.emotion-9 {
  max-width: 25rem;
  width: auto;
}

.emotion-5 {
  display: block;
  position: relative;
}

.emotion-3 select,
.emotion-3 input {
  border-radius: 100px;
}

.emotion-1 {
  display: block;
  margin-bottom: 0.5rem;
  color: #8697A2;
  font-size: 1rem;
}

.emotion-2 {
  width: 100%;
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  color: #FFF;
  background-color: #31373F;
  border: 1px solid #31373F;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
}

.emotion-2:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.emotion-2:not(:focus):not(:placeholder-shown):invalid {
  border-color: #FF4E64;
}

.emotion-2:invalid,
.emotion-2:-moz-ui-invalid {
  box-shadow: none;
}

.emotion-2:focus {
  border-color: #8697A2;
  outline: none;
}

.emotion-4 {
  position: absolute;
  right: 0.5rem;
  bottom: 0.5rem;
}

<form
  className="emotion-9"
>
  <fieldset
    className="emotion-8"
  >
    <legend
      className="emotion-0"
    >
      2FA
    </legend>
    <ul
      className="emotion-7"
    >
      <li
        className="emotion-6"
      >
        <div
          className="emotion-5"
        >
          <div
            className="emotion-3"
          >
            <label>
              <span
                className="emotion-1"
              >
                Enable Two Factor Authentication
              </span>
              <input
                className="emotion-2"
                placeholder="Enter code"
                type="text"
              />
            </label>
          </div>
          <div
            className="emotion-4"
          >
            <button
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </li>
    </ul>
  </fieldset>
</form>
`;

exports[`Form renders a form combo without a label 1`] = `
.emotion-7 {
  border: 0;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-6 {
  display: block;
  margin-left: 0;
  margin-top: 0;
}

.emotion-5 {
  display: block;
  list-style-type: none;
  margin-bottom: 1rem;
}

.emotion-5:before {
  position: absolute;
  content: "\\200B";
}

.emotion-8 {
  max-width: 25rem;
  width: auto;
}

.emotion-4 {
  display: block;
  position: relative;
}

.emotion-2 select,
.emotion-2 input {
  border-radius: 100px;
}

.emotion-1 {
  width: 100%;
  max-width: 100%;
  padding: 0.875rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font: inherit;
  font-size: 1rem;
  color: #FFF;
  background-color: #31373F;
  border: 1px solid #31373F;
  border-radius: 4px;
  -webkit-transition: border-color 0.2s ease;
  transition: border-color 0.2s ease;
  -webkit-appearance: none;
}

.emotion-1:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.emotion-1:not(:focus):not(:placeholder-shown):invalid {
  border-color: #FF4E64;
}

.emotion-1:invalid,
.emotion-1:-moz-ui-invalid {
  box-shadow: none;
}

.emotion-1:focus {
  border-color: #8697A2;
  outline: none;
}

.emotion-3 {
  position: absolute;
  right: 0.5rem;
  bottom: 0.5rem;
}

<form
  className="emotion-8"
>
  <fieldset
    className="emotion-7"
  >
    <legend
      className="emotion-0"
    >
      2FA
    </legend>
    <ul
      className="emotion-6"
    >
      <li
        className="emotion-5"
      >
        <div
          className="emotion-4"
        >
          <div
            className="emotion-2"
          >
            <input
              className="emotion-1"
              placeholder="Enter code"
              type="text"
            />
          </div>
          <div
            className="emotion-3"
          >
            <button
              type="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </li>
    </ul>
  </fieldset>
</form>
`;

exports[`Form renders correctly 1`] = `
.emotion-9 {
  width: auto;
}

.emotion-8 {
  border: 0;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-7 {
  display: block;
  margin-left: 0;
  margin-top: 0;
}

.emotion-1 {
  display: block;
  list-style-type: none;
  margin-bottom: 1rem;
}

.emotion-1:before {
  position: absolute;
  content: "\\200B";
}

.emotion-3 {
  display: inline-block;
  margin-top: 0.5rem;
  font-size: 0.75rem;
  color: #FF4E64;
}

.emotion-2 {
  display: inline-block;
  width: 0.875rem;
  height: 0.875rem;
  margin-right: 0.5rem;
  vertical-align: sub;
  background-position: center;
  background-size: 60%,100%;
  background-repeat: no-repeat;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 4 20' xmlns='http://www.w3.org/2000/svg' color='white' class='css-uk6cul'%3e%3cpath d='M4 12H0V0h4v12zm0 8H0v-4h4v4z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e"), url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg' color='%23FF4E64' class='css-uk6cul'%3e%3ccircle cx='10' cy='10' r='10' fill='currentColor' fill-rule='evenodd'%3e%3c/circle%3e%3c/svg%3e");
}

<form
  className="emotion-9"
>
  <fieldset
    className="emotion-8"
  >
    <legend
      className="emotion-0"
    >
      Sign up
    </legend>
    <ul
      className="emotion-7"
    >
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Name
          <input
            id="testInput1"
            placeholder="e.g. John Appleseed"
            type="text"
          />
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Password
          <input
            id="testInput1"
            type="password"
          />
        </label>
        <p
          className="emotion-3"
          role="alert"
        >
          <i
            className="emotion-2"
          />
          Please enter a valid password.
        </p>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput2"
        >
          <input
            id="testInput2"
            name="terms"
            type="checkbox"
            value="terms"
          />
          I agree to the terms and conditions.
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <button
          type="submit"
          value="submit"
        >
          Sign up
        </button>
      </li>
    </ul>
  </fieldset>
</form>
`;

exports[`Form renders with HTML form attributes 1`] = `
.emotion-9 {
  width: auto;
}

.emotion-8 {
  border: 0;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-7 {
  display: block;
  margin-left: 0;
  margin-top: 0;
}

.emotion-1 {
  display: block;
  list-style-type: none;
  margin-bottom: 1rem;
}

.emotion-1:before {
  position: absolute;
  content: "\\200B";
}

.emotion-3 {
  display: inline-block;
  margin-top: 0.5rem;
  font-size: 0.75rem;
  color: #FF4E64;
}

.emotion-2 {
  display: inline-block;
  width: 0.875rem;
  height: 0.875rem;
  margin-right: 0.5rem;
  vertical-align: sub;
  background-position: center;
  background-size: 60%,100%;
  background-repeat: no-repeat;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 4 20' xmlns='http://www.w3.org/2000/svg' color='white' class='css-uk6cul'%3e%3cpath d='M4 12H0V0h4v12zm0 8H0v-4h4v4z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e"), url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg' color='%23FF4E64' class='css-uk6cul'%3e%3ccircle cx='10' cy='10' r='10' fill='currentColor' fill-rule='evenodd'%3e%3c/circle%3e%3c/svg%3e");
}

<form
  action="#"
  autocomplete="on"
  className="emotion-9"
  method="post"
>
  <fieldset
    className="emotion-8"
  >
    <legend
      className="emotion-0"
    >
      Sign up
    </legend>
    <ul
      className="emotion-7"
    >
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Name
          <input
            id="testInput1"
            placeholder="e.g. John Appleseed"
            type="text"
          />
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Password
          <input
            id="testInput1"
            type="password"
          />
        </label>
        <p
          className="emotion-3"
          role="alert"
        >
          <i
            className="emotion-2"
          />
          Please enter a valid password.
        </p>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput2"
        >
          <input
            id="testInput2"
            name="terms"
            type="checkbox"
            value="terms"
          />
          I agree to the terms and conditions.
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <button
          type="submit"
          value="submit"
        >
          Sign up
        </button>
      </li>
    </ul>
  </fieldset>
</form>
`;

exports[`Form renders with custom fullWidth 1`] = `
.emotion-8 {
  border: 0;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-7 {
  display: block;
  margin-left: 0;
  margin-top: 0;
}

.emotion-1 {
  display: block;
  list-style-type: none;
  margin-bottom: 1rem;
}

.emotion-1:before {
  position: absolute;
  content: "\\200B";
}

.emotion-3 {
  display: inline-block;
  margin-top: 0.5rem;
  font-size: 0.75rem;
  color: #FF4E64;
}

.emotion-2 {
  display: inline-block;
  width: 0.875rem;
  height: 0.875rem;
  margin-right: 0.5rem;
  vertical-align: sub;
  background-position: center;
  background-size: 60%,100%;
  background-repeat: no-repeat;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 4 20' xmlns='http://www.w3.org/2000/svg' color='white' class='css-uk6cul'%3e%3cpath d='M4 12H0V0h4v12zm0 8H0v-4h4v4z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e"), url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg' color='%23FF4E64' class='css-uk6cul'%3e%3ccircle cx='10' cy='10' r='10' fill='currentColor' fill-rule='evenodd'%3e%3c/circle%3e%3c/svg%3e");
}

.emotion-9 {
  width: 100%;
}

<form
  className="emotion-9"
>
  <fieldset
    className="emotion-8"
  >
    <legend
      className="emotion-0"
    >
      Sign up
    </legend>
    <ul
      className="emotion-7"
    >
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Name
          <input
            id="testInput1"
            placeholder="e.g. John Appleseed"
            type="text"
          />
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Password
          <input
            id="testInput1"
            type="password"
          />
        </label>
        <p
          className="emotion-3"
          role="alert"
        >
          <i
            className="emotion-2"
          />
          Please enter a valid password.
        </p>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput2"
        >
          <input
            id="testInput2"
            name="terms"
            type="checkbox"
            value="terms"
          />
          I agree to the terms and conditions.
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <button
          type="submit"
          value="submit"
        >
          Sign up
        </button>
      </li>
    </ul>
  </fieldset>
</form>
`;

exports[`Form renders with custom maxWidth 1`] = `
.emotion-8 {
  border: 0;
}

.emotion-0 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.emotion-7 {
  display: block;
  margin-left: 0;
  margin-top: 0;
}

.emotion-1 {
  display: block;
  list-style-type: none;
  margin-bottom: 1rem;
}

.emotion-1:before {
  position: absolute;
  content: "\\200B";
}

.emotion-3 {
  display: inline-block;
  margin-top: 0.5rem;
  font-size: 0.75rem;
  color: #FF4E64;
}

.emotion-2 {
  display: inline-block;
  width: 0.875rem;
  height: 0.875rem;
  margin-right: 0.5rem;
  vertical-align: sub;
  background-position: center;
  background-size: 60%,100%;
  background-repeat: no-repeat;
  background-image: url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 4 20' xmlns='http://www.w3.org/2000/svg' color='white' class='css-uk6cul'%3e%3cpath d='M4 12H0V0h4v12zm0 8H0v-4h4v4z' fill='currentColor' fill-rule='nonzero'%3e%3c/path%3e%3c/svg%3e"), url("data:image/svg+xml,%3csvg width='1em' height='1em' viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg' color='%23FF4E64' class='css-uk6cul'%3e%3ccircle cx='10' cy='10' r='10' fill='currentColor' fill-rule='evenodd'%3e%3c/circle%3e%3c/svg%3e");
}

.emotion-9 {
  max-width: 30rem;
  width: auto;
}

<form
  className="emotion-9"
>
  <fieldset
    className="emotion-8"
  >
    <legend
      className="emotion-0"
    >
      Sign up
    </legend>
    <ul
      className="emotion-7"
    >
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Name
          <input
            id="testInput1"
            placeholder="e.g. John Appleseed"
            type="text"
          />
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput1"
        >
          Password
          <input
            id="testInput1"
            type="password"
          />
        </label>
        <p
          className="emotion-3"
          role="alert"
        >
          <i
            className="emotion-2"
          />
          Please enter a valid password.
        </p>
      </li>
      <li
        className="emotion-1"
      >
        <label
          htmlFor="testInput2"
        >
          <input
            id="testInput2"
            name="terms"
            type="checkbox"
            value="terms"
          />
          I agree to the terms and conditions.
        </label>
      </li>
      <li
        className="emotion-1"
      >
        <button
          type="submit"
          value="submit"
        >
          Sign up
        </button>
      </li>
    </ul>
  </fieldset>
</form>
`;
